### PR TITLE
{math}[foss/2019a] thurstonianIRT v0.9.0 w/ R 3.6.0

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
needs work for the `intel` easyconfig, compilation of `thurstonianIRT` fails with:

```
/software/R/3.6.0-intel-2019a/lib64/R/library/StanHeaders/include/stan/math/rev/scal/meta/operands_and_partials.hpp(70): error: more than one partial specialization matches the template argument list of class "stan::math::internal::ops_partials_edge<double, Eigen::Matrix<stan::math::var, -1, 1, 0, -1, 1>>"
            "stan::math::internal::ops_partials_edge<double, Eigen::Matrix<stan::math::var, R, C, <expression>, R, C>>"
            "stan::math::internal::ops_partials_edge<ViewElt, Eigen::Matrix<Op, R, C, <expression>, R, C>>"
    internal::ops_partials_edge<double, Op1> edge1_;
```

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2089.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/981a2912bd941791c503816e5f39150a for a full test report.

**Comment by boegel** [comment]
For future reference, here's how the compilation fails with Intel compilers (19.0.1, `intel/2019a`):

```
/software/R/3.6.0-intel-2019a/lib64/R/library/StanHeaders/include/stan/math/rev/scal/meta/operands_and_partials.hpp(70): error: more than one partial specialization matches the template argument list of class "stan::math::internal::ops_partials_edge<double, Eigen::Matrix<stan::math::var, -1, 1, 0, -1, 1>>"
            "stan::math::internal::ops_partials_edge<double, Eigen::Matrix<stan::math::var, R, C, <expression>, R, C>>"
            "stan::math::internal::ops_partials_edge<ViewElt, Eigen::Matrix<Op, R, C, <expression>, R, C>>"
    internal::ops_partials_edge<double, Op1> edge1_;
                                             ^
          detected during:
            instantiation of class "stan::math::operands_and_partials<Op1, Op2, Op3, Op4, Op5, stan::math::var> [with Op1=Eigen::Matrix<stan::math::var, -1, 1, 0, -1, 1>, Op2=double, Op3=double, Op4=double, Op5=double]" at line 63 of "/software/R/3.6.0-intel-2019a/lib64/R/library/StanHeaders/include/stan/math/prim/scal/prob/bernoulli_lpmf.hpp"
            instantiation of "stan::return_type<T_prob>::type stan::math::bernoulli_lpmf<propto,T_n,T_prob>(const T_n &, const T_prob &) [with propto=true, T_n=std::vector<int, std::allocator<int>>, T_prob=Eigen::Matrix<stan::math::var, -1, 1, 0, -1, 1>]" at line 16 of "/software/R/3.6.0-intel-2019a/lib64/R/library/StanHeaders/include/stan/math/prim/scal/prob/bernoulli_log.hpp"
            instantiation of "stan::return_type<T_prob>::type stan::math::bernoulli_log<propto,T_n,T_prob>(const T_n &, const T_prob &) [with propto=true, T_n=std::vector<int, std::allocator<int>>, T_prob=Eigen::Matrix<stan::math::var, -1, 1, 0, -1, 1>]" at line 982 of "stan_files/thurstonian_irt_model.hpp"
            instantiation of "T__ model_thurstonian_irt_model_namespace::model_thurstonian_irt_model::log_prob<propto__,jacobian__,T__>(std::vector<T__, std::allocator<T__>> &, std::vector<int, std::allocator<int>> &, std::ostream *) const [with propto__=true, jacobian__=true, T__=stan::math::var]" at line 46 of "/software/R/3.6.0-intel-2019a/lib64/R/library/StanHeaders/include/src/stan/model/log_prob_propto.hpp"
            instantiation of "double stan::model::log_prob_propto<jacobian_adjust_transform,M>(const M &, std::vector<double, std::allocator<double>> &, std::vector<int, std::allocator<int>> &, std::ostream *) [with jacobian_adjust_transform=true, M=stan_model]" at line 1122 of "/software/R/3.6.0-intel-2019a/lib64/R/library/rstan/include/rstan/stan_fit.hpp"
            instantiation of "SEXP rstan::stan_fit<Model, RNG_t>::log_prob(SEXP, SEXP, SEXP) [with Model=stan_model, RNG_t=boost::random::ecuyer1988]" at line 22 of "stan_files/thurstonian_irt_model.cc"

compilation aborted for stan_files/thurstonian_irt_model.cc (code 2)
```

Looks like this is a known but not yet resolved problems in the Intel compilers, see https://software.intel.com/en-us/forums/intel-c-compiler/topic/781749.

To escape this hot mess, I'll resort to creating a separate bundle for `thurstonianIRT` and the missing deps using only `foss/2019a`...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/df801e3ae22e0a4a7c6c557b511878a8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2710.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a11885d6d1cbf6c00f3ed0fc2d8d781b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2106.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ca213c74733e81accd910082adbce18d for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node371.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/6b4d8f0d7bc3d104876c2a1c78730c57 for a full test report.

File: easybuild/easyconfigs/t/thurstonianIRT/thurstonianIRT-0.9.0-foss-2019a-R-3.6.0.eb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+exts_list = [
+    ('StanHeaders', '2.19.0', {
+        'checksums': ['a0ab8f996a7a903a37491f6e4e8cf8b9f51bf0c13a5326171fab90692fe14592'],
+    }),
+    ('loo', '2.1.0', {
+        'checksums': ['1bf4a1ef85d151577ff96d4cf2a29c9ef24370b0b1eb08c70dcf45884350e87d'],
+    }),
+    ('rstan', '2.19.2', {
```
Comment (smoors) [inline easybuild/easyconfigs/t/thurstonianIRT/thurstonianIRT-0.9.0-foss-2019a-R-3.6.0.eb]: FYI: there is already separate easyconfig for RStan.

**Comment by smoors** [comment]
hugging travis

**Comment by smoors** [comment]
Going in, thanks @boegel!

**Pull request merged**